runtime/resolvers/sql.go

```diff
@@ -19,121 +18,146 @@ import (
+	// NOTE: Not exhaustive. Any other args are passed to the "args" property when resolving the SQL template.
+}
+
+// newSQL creates a resolver that executes a SQL query.
+// It supports the use of templating in the SQL string to inject user attributes and args into the SQL query.
+func newSQL(ctx context.Context, opts *runtime.ResolverOptions) (runtime.Resolver, error) {
+	return newSQLWithRefs(ctx, opts, nil)
+}
+
+// newSQLWithRefs is similar to newSQL, but allows overriding the resolver's refs.
+// This capability is required by the metrics SQL resolver to wrap this regular SQL resolver.
+func newSQLWithRefs(ctx context.Context, opts *runtime.ResolverOptions, refsOverride []*runtimev1.ResourceName) (runtime.Resolver, error) {
+	props := &sqlProps{}
+	if err := mapstructure.Decode(opts.Properties, props); err != nil {
+		return nil, err
 	}
-	resolvedSQL, deps, err := resolveSQLAndDeps(ctx, sql, opts)
-	if err != nil {
+
+	args := &sqlArgs{}
+	if err := mapstructure.Decode(opts.Args, args); err != nil {
 		return nil, err
 	}
+
 	olap, release, err := opts.Runtime.OLAP(ctx, opts.InstanceID)
 	if err != nil {
 		return nil, err
 	}
 
-	return &SQLResolver{
-		resolvedSQL: resolvedSQL,
-		deps:        deps,
+	resolvedSQL, refs, err := buildSQL(props.SQL, olap.Dialect(), opts.Args, opts.UserAttributes, opts.ForExport)
+	if err != nil {
+		return nil, err
+	}
+
+	if refsOverride != nil {
+		refs = refsOverride
+	}
+
+	return &sqlResolver{
+		sql:         resolvedSQL,
+		refs:        refs,
 		olap:        olap,
-		releaseFunc: release,
+		olapRelease: release,
+		priority:    args.Priority,
 	}, nil
 }
 
-// Key that can be used for caching
-func (r *SQLResolver) Key() string {
-	return r.resolvedSQL
+func (r *sqlResolver) Close() error {
+	r.olapRelease()
+	return nil
+}
+
+func (r *sqlResolver) Key() string {
+	return r.sql
 }
 
-// Deps referenced by the query
-func (r *SQLResolver) Deps() []*runtimev1.ResourceName {
-	return r.deps
+func (r *sqlResolver) Refs() []*runtimev1.ResourceName {
+	return r.refs
 }
 
-// Validate the query without running any "expensive" operations
-func (r *SQLResolver) Validate(ctx context.Context) error {
+func (r *sqlResolver) Validate(ctx context.Context) error {
 	_, err := r.olap.Execute(ctx, &drivers.Statement{
-		Query:  r.resolvedSQL,
+		Query:  r.sql,
 		DryRun: true,
 	})
 	return err
 }
 
-// ResolveInteractive Resolve for interactive use (e.g. API requests or alerts)
-func (r *SQLResolver) ResolveInteractive(ctx context.Context, priority int) (runtime.Result, error) {
+func (r *sqlResolver) ResolveInteractive(ctx context.Context) (*runtime.ResolverResult, error) {
 	res, err := r.olap.Execute(ctx, &drivers.Statement{
-		Query:    r.resolvedSQL,
-		Priority: priority,
+		Query:    r.sql,
+		Priority: r.priority,
 	})
 	if err != nil {
-		return runtime.Result{}, err
+		return nil, err
 	}
 	defer res.Close()
 
-	var out []map[string]interface{}
+	var out []map[string]any
```

Comment:
trying to understand, what difference will this make ?

Reply:
It's the same, just syntactic sugar